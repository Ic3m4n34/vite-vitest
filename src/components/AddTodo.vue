<template>
  <div
    class="flex flex-row space-x-2"
  >
    <input
      v-model="todoInput"
      type="text"
      placeholder="Add a todo"
      class="border border-gray-300 rounded-lg py-2 px-4 text-gray-800"
      @keyup.enter="addTodo"
    >
    <button
      class="bg-purple-700 text-white py-2 px-4 rounded-lg"
      @click="addTodo"
    >
      Add Todo
    </button>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref } from 'vue';
import { nanoid } from 'nanoid';

export default defineComponent({
  name: 'AddTodo',
  emits: ['task-added'],
  setup(props, { emit }) {
    const todoInput = ref('');

    const addTodo = () => {
      emit('task-added', { todo: todoInput.value, id: nanoid(), done: false });
      todoInput.value = '';
    };

    return {
      todoInput,
      addTodo,
    };
  },
});
</script>

<style scoped>

</style>
