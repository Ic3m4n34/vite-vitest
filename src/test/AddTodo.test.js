/**
 * @vitest-environment jsdom
 */

import { mount } from '@vue/test-utils';
import {
  beforeEach, describe, it, expect,
} from 'vitest';
import AddTodo from '../components/AddTodo.vue';

let wrapper;

beforeEach(() => {
  wrapper = mount(AddTodo);
});

describe('AddTodo', () => {
  describe('rendering', () => {
    it('renders the component', () => {
      expect(wrapper.exists()).toBe(true);
    });

    it('matches snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
