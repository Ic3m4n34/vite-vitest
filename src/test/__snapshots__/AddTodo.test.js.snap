// Vitest Snapshot v1

exports[`AddTodo > rendering > matches snapshot 1`] = `"<div class=\\"flex flex-row space-x-2\\"><input type=\\"text\\" placeholder=\\"Add a todo\\" class=\\"border border-gray-300 rounded-lg py-2 px-4 text-gray-800\\"><button class=\\"bg-purple-700 text-white py-2 px-4 rounded-lg\\"> Add Todo </button></div>"`;

exports[`rendering > matches snapshot 1`] = `"<div class=\\"flex flex-row space-x-2\\"><input type=\\"text\\" placeholder=\\"Add a todo\\" class=\\"border border-gray-300 rounded-lg py-2 px-4 text-gray-800\\"><button class=\\"bg-purple-700 text-white py-2 px-4 rounded-lg\\"> Add Todo </button></div>"`;
